src/radical/pilot/agent/scheduler/flux.py

```diff
@@ -0,0 +1,148 @@
+
+__copyright__ = "Copyright 2017, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import json
+
+import radical.utils        as ru
+
+from ...   import states    as rps
+from ...   import constants as rpc
+
+from ..lm  import LaunchMethod
+from .base import AgentSchedulingComponent
+
+
+# ------------------------------------------------------------------------------
+#
+class Flux(AgentSchedulingComponent):
+    '''
+    Pass all scheduling and execution control to Flux, and leave it to the
+    execution component to collect state updates.
+    '''
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, session):
+
+        self.nodes = None
+
+        AgentSchedulingComponent.__init__(self, cfg, session)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        import flux
+
+        flux_url   = self._cfg['lrms_info']['lm_info']['flux_env']['FLUX_URI']
+        self._flux = flux.Flux(url=flux_url)
+
+        # don't advance tasks via the component's `advance()`, but push them
+        # toward the executor *without state change* - state changes are
+        # performed in retrospect by the executor, based on the scheduling and
+        # execution events collected from Flux.
+        qname   = rpc.AGENT_EXECUTING_QUEUE
+        fname   = '%s/%s.cfg' % (self._cfg.path, qname)
+        cfg     = ru.read_json(fname)
+        self._q = ru.zmq.Putter(qname, cfg['put'])
+
+        # create job spec via the flux LM
+        self._lm = LaunchMethod.create(name    = 'FLUX',
+                                       cfg     = self._cfg,
+                                       session = self._session)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def work(self, units):
+
+        # overload the base class work method
+
+        from flux import job as flux_job
+
+        self.advance(units, rps.AGENT_SCHEDULING, publish=True, push=False)
+
+        for unit in units:
+
+          # # FIXME: transfer from executor
+          # self._cu_environment = self._populate_cu_environment()
+
+            jd  = json.dumps(ru.read_json('/home/merzky/projects/flux/spec.json'))
```

Comment:
add example, maybe initial translation, check GPUs

Reply:
just to ping this comment (in case it is still relevant)

Reply:
thanks for the ping, will do.